tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -11173,11 +11173,12 @@ SpirvInstruction *SpirvEmitter::processIntrinsicUsingSpirvInst(
     case spv::Op::OpFwidth:
     case spv::Op::OpFwidthFine:
     case spv::Op::OpFwidthCoarse:
+      if (spvContext.isCS())
```

Comment:
I'm a bit confused. The existing comment above this block looks contradictory:

```
  // Certain opcodes are only allowed in pixel shader
  if (!spvContext.isPS())
```

and for a few of the opcodes I checked, the spec does say "This instruction is only valid in the Fragment Execution Model." ([ref](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_derivative_instructions)) which would make the comment correct and our generated SPIR-V invalid. Correcting it would also void this new addition though. Do you have any more context for how these functions should be used?

Reply:
I forgot to update the comment. The extension allows these opcodes to be used in compute shaders as well. I'll update the comment.

Reply:
I updated the comment to say:

>   // The derivative opcodes are only allowed in pixel shader, or in compute
  // shaderers when the SPV_NV_compute_shader_derivatives is enabled.

Let me know if that makes sense.

Reply:
The comment looks correct, but the condition is `if (!spvContext.isPS())` (emphasis on the **!**). Is that an existing bug?

Reply:
Got it, thanks for the comment updates, LGTM!